modules/codelite/codelite_project.lua

```diff
@@ -207,7 +207,10 @@
 			_x(4, '<IncludePath Value="%s"/>', project.getrelative(cfg.project, includedir))
 		end
 		for _, define in ipairs(cfg.defines) do
-			_x(4, '<Preprocessor Value="%s"/>', p.esc(define))
+			-- All characters encoded for XML (& -> &amp;) need to be escaped with backslash,
+			-- same applies to space. This was correct for CodeLite 12 and took a lot of trial
+			-- and error to verify.
+			_p(4, '<Preprocessor Value="%s"/>', p.esc(define):gsub('&', '\\&'):gsub(' ', '\\ '))
```

Comment:
Is it intended on the CodeLite side to have different escaping depending on the field in their project file?
This feel more like an issue to me but I maybe I'm wrong.

Reply:
Honestly, I have no idea. This one field seemed to have serious issues with everything, and caused the cURL project to not load. It seems like CodeLite converts the escaped values back, and passes the value to the compiler. If you don't use `\` on the `"`, strings come through without quotes and are treated as identifiers and builds fail. Having `\"` instead seems to cause `&`, `<`, `>` and ` ` to be treated like normal on the command line, so they also needed to be escaped to prevent builds failing, or random files being created on disk.

I'll see how these magic characters are handled in `gmake2` since both Make and CodeLite just pass params straight to the compiler. I'll then adjust this accordingly - last night I just kept iterating and didn't really consider what the other generators did to handle this craziness.

Reply:
Looks like `gmake2` will only escape `"` and ` ` with a `\`. Builds fail if you have `&`, `<`, or `>` in `defines` without escaping them, I guess I could change the `"` escape to `\&quot;` which is what it used to be and then add ` ` -> `\ `. I'll see how that works out.

Reply:
Fixed.